Add CHANGELOG for release 0.2

pxLi opened the pull request:
Signed-off-by: Peixin Li <pxli@nyu.edu>

According to the output there are a few PR w/o `project` info, 
@sameerz @jlowe @revans2 @tgravescs Please help review the changelog

```bash
# Run w/ generate-changelog
GITHUB_TOKEN=xxx  scripts/generate-changelog --releases=0.2,0.1

# Output:
Generating changelog ...
Processing pull requests ...
Processing issues ...
Done.
Merged Pull Requests w/o Project:
branch-0.1 #181 Remove INCOMPAT for NormalizeNanAndZero, KnownFloatingPointNormalized https://github.com/NVIDIA/spark-rapids/pull/181
branch-0.1 #191 WindowExec handle different spark distributions https://github.com/NVIDIA/spark-rapids/pull/191
branch-0.1 #195 Support public release for plugin jar https://github.com/NVIDIA/spark-rapids/pull/195
branch-0.1 #206 Change groupID to 'com.nvidia' in IT scripts https://github.com/NVIDIA/spark-rapids/pull/206
branch-0.1 #208 Remove unneeded comment from pom.xml https://github.com/NVIDIA/spark-rapids/pull/208
branch-0.2 #645 [window] Add GpuWindowExec requiredChildOrdering https://github.com/NVIDIA/spark-rapids/pull/645
branch-0.2 #714 Move spark 3.0.1-shims out of snapshot-shims https://github.com/NVIDIA/spark-rapids/pull/714
```

FYI
![image](https://user-images.githubusercontent.com/8086184/93039159-5d98eb80-f679-11ea-9f9c-e213d6179b99.png)
To get a preview of the Markdown file

jlowe commented:
#645  and #714 were merged to branch-0.2 but did not have 0.2 set as a project and were missed from the changelog.  It thought the changelog tool was supposed to report PRs merged without a project?

jlowe commented:
I've updated the project on those two PRs.

pxLi commented:
> #645 and #714 were merged to branch-0.2 but did not have 0.2 set as a project and were missed from the changelog. It thought the changelog tool was supposed to report PRs merged without a project?

Yes, it does not list PR w/o project. And it reports the info in stdout for the tool user.

I will recommit a new one after https://github.com/NVIDIA/spark-rapids/pull/723 get merged

jlowe commented on CHANGELOG.md:
```diff
@@ -0,0 +1,381 @@
+|[#255](https://github.com/NVIDIA/spark-rapids/pull/255)|Revert change to make dependencies of shaded jar optional|
+|[#257](https://github.com/NVIDIA/spark-rapids/pull/257)|Fix link to RAPIDS cudf in index.md|
+|[#252](https://github.com/NVIDIA/spark-rapids/pull/252)|Update to 0.2.0-SNAPSHOT and cudf-0.15-SNAPSHOT|
+## Release 0.1
```
It would be nice to add some whitespace before this line for those reading this as a regular text file.

jlowe commented on CHANGELOG.md:
```diff
@@ -0,0 +1,381 @@
+|[#74](https://github.com/NVIDIA/spark-rapids/issues/74)|[FEA] Support ToUnixTimestamp|
+|[#21](https://github.com/NVIDIA/spark-rapids/issues/21)|[FEA] NormalizeNansAndZeros|
+|[#105](https://github.com/NVIDIA/spark-rapids/issues/105)|[FEA] integration tests for equi-joins|
+### Bugs Fixed
```
Similarly nice to have whitespace before these and other header lines when viewing as text.

jlowe commented:
#479 was not implemented, I marked it as `wontfix`.  I also fixed typos in a number of issue/PR headlines.

#579 is an issue in the 0.2 release that is missing from the changelog.  Please investigate how that was missed, there may be others missing as well.  I'm guessing because it is missing the `feature request`, label, so I just added it to that one.  However there may be others.

pxLi commented:
> #479 was not implemented, I marked it as `wontfix`. I also fixed typos in a number of issue/PR headlines.
> 
> #579 is an issue in the 0.2 release that is missing from the changelog. Please investigate how that was missed, there may be others missing as well. I'm guessing because it is missing the `feature request`, label, so I just added it to that one. However there may be others.

Thx, #579 is added when I run script again

sameerz commented:
The issues on branch-0.1 were part of the 0.1 release, I moved them to that project.

pxLi commented:
> The PRs on branch-0.1 were part of the 0.1 release, I moved them to that project.

Thx, looks OK from local run. I will wait for https://github.com/NVIDIA/spark-rapids/pull/723 premerge run

sameerz commented:
#723 is going to go into 0.3 (see the project update in the PR from two days ago).  @revans2 can you retarget the PR to branch-0.3?

pxLi commented:
> #723 is going to go into 0.3 (see the project update in the PR from two days ago). @revans2 can you retarget the PR to branch-0.3?

kk, I just push a lastest changelog

pxLi commented:
build

pxLi commented:
build

pxLi commented:
@jlowe @sameerz Let me know if the latest changelog is OK or not, the release phase is blocked by this PR. thx!

sameerz reviewed (approved):
Look good to me
